compiler/src/dmd/iasmgcc.d

```diff
@@ -400,38 +448,49 @@ GccAsmStatement parseGccAsm(Parser)(Parser p, GccAsmStatement s)
     if (s.insn.isErrorExp())
         return s;
 
+    // No semicolon followed after instruction template, treat as extended asm.
     if (p.token.value == TOK.semicolon || p.token.value == TOK.endOfFile)
         goto Ldone;
 
-    // No semicolon followed after instruction template, treat as extended asm.
-    foreach (section; 0 .. 4)
-    {
-        p.check(TOK.colon);
-
-        final switch (section)
-        {
-            case 0:
-                s.outputargs = p.parseExtAsmOperands(s);
-                break;
-
-            case 1:
-                p.parseExtAsmOperands(s);
-                break;
+    // Look for outputs.
+    if (p.requireToken(TOK.colon))
+        s.outputargs = p.parseExtAsmOperands(s);
+    else
+        goto Ldone;
```

Comment:
Unrolled loop.  Now only the first `:` is required for extended syntax.  All others are optional, and instead an error is emitted for an expected `;` token.